pkg/client/testdata/ssh.golden

```diff
@@ -35,72 +35,38 @@ metadata:
 ---
 apiVersion: v1
 data:
-  e2e.yaml: |
+  plugin-0.yaml: |
+    extra-volumes:
+    - name: sshkey-vol
+      secret:
+        defaultMode: 256
+        secretName: ssh-key
     sonobuoy-config:
       driver: Job
       plugin-name: e2e
       result-type: e2e
     spec:
+      command:
+      - /run_e2e.sh
       env:
       - name: E2E_FOCUS
-        value: ''
       - name: E2E_SKIP
-        value: ''
       - name: E2E_PARALLEL
-        value: ''
       - name: LOCAL_SSH_KEY
-        value: 'id_rsa'
+        value: id_rsa
       - name: AWS_SSH_KEY
-        value: '/root/.ssh/id_rsa'
+        value: /root/.ssh/id_rsa
       - name: KUBE_SSH_KEY
-        value: 'id_rsa'
+        value: id_rsa
       - name: KUBE_SSH_USER
         value: ssh-user
-      command: ["/run_e2e.sh"]
-      image: 
-      imagePullPolicy: 
       name: e2e
+      resources: {}
       volumeMounts:
       - mountPath: /tmp/results
         name: results
-        readOnly: false
       - mountPath: /root/.ssh
         name: sshkey-vol
-      tolerations:
```

Comment:
The tolerations is one thing worth noting: There is no field on the object `tolerations` so this just gets dropped. The YAML we now generate is accurate: serialized by the standard library and types are from the k8s vendored code.